code/game/objects/structures/secure_safe.dm

```diff
@@ -11,16 +11,23 @@
 	base_icon_state = "wall_safe"
 	anchored = TRUE
 	density = FALSE
+	/// Whether or not this safe will be deconstructed if the wall it is on is deconstructed.
+	var/deconstruct_on_wall_deconstruct = TRUE
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/structure/secure_safe, 32)
 
 /obj/structure/secure_safe/Initialize(mapload)
 	. = ..()
 	//this will create the storage for us.
 	AddComponent(/datum/component/lockable_storage)
-	find_and_hang_on_wall()
+	find_and_hang_on_wall(custom_drop_callback = CALLBACK(src, PROC_REF(handle_wall_deconstruct)))
 	PopulateContents()
 
+/obj/structure/secure_safe/proc/handle_wall_deconstruct()
+	if(!deconstruct_on_wall_deconstruct)
+		return
+	deconstruct(disassembled = FALSE)
```

Comment:
```suggestion
	set_anchored(FALSE)
```

It now becomes an item you can drag around (but can't pick up which is intended) See


https://github.com/tgstation/tgstation/assets/110812394/bf6ec69a-0e12-4ff5-a38b-b7195f8af891